daemons/controld/controld_execd.c

```diff
@@ -1975,6 +1978,18 @@ construct_op(lrm_state_t *lrm_state, xmlNode *rsc_op, const char *rsc_id,
         op->interval_ms = 0;
     }
 
+    /* Use pcmk_monitor_timeout instead of meta timeout for stonith
+       recurring monitor */
+    primitive = find_xml_node(rsc_op, XML_CIB_TAG_RESOURCE, FALSE);
+    class = crm_element_value(primitive, XML_AGENT_ATTR_CLASS);
+
+    if (is_set(pcmk_get_ra_caps(class), pcmk_ra_cap_fence_params)
+            && pcmk__str_eq(operation, CRMD_ACTION_STATUS, pcmk__str_casei)
+            && (op->interval_ms > 0)) {
+        op_timeout = g_hash_table_lookup(params, "pcmk_monitor_timeout");
+        op->timeout = crm_get_msec(op_timeout);
```

Comment:
This needs to be changed. It overwrites the meta timeout even if there is no pcmk_monitor_timeout.